types/isarray/index.d.ts

```diff
@@ -1,3 +1,4 @@
 export = isarray;
 
-declare function isarray(value: unknown): value is unknown[];
+// eslint-disable-next-line @definitelytyped/no-unnecessary-generics
+declare function isarray<T>(value: unknown): value is T[];
```

Comment:
This seems pretty suspicious. This means you can instantiate this function with any type argument and it'll look like it works, but it'll be a lie. (And that's what the lint rule is mad)

Reply:
Hmm - the lint rule seems to be just about not using a generic only once, but there’s lots of valid use cases for that, and the purpose here is to ensure that TS can type-narrow down to an array, which is what this change accomplishes. Is there a better way to do it?

Reply:
I'm not sure what valid use cases you mean; usually it's only to the effect of lying to the type system, like ORMs or something.

Is the broken case just the test in the PR?

Given this is effectively a polyfill for the real isArray, I would really be just reusing its types.

(I can't really test any of this for another week; I am far from home without a way to check)

Reply:
No rush then :-) basically yeah, TS isn’t currently narrowing properly for this package. I’ll check if the real types achieve this, but if not, should i be updating those too?

Reply:
isArray has a long history of everyone wanting it to behave differently, https://github.com/microsoft/TypeScript/issues/17002 among others

There's upsides and downsides to every formation :/